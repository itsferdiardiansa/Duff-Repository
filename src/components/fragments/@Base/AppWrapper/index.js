/* eslint-disable no-unused-vars */
import {
  defineComponent,
  h,
  ref,
  getCurrentInstance,
  unref,
  computed,
} from 'vue'
import { useStore } from 'vuex'
import { useRouter } from 'vue-router'
import globalSetting from '@plugin/globalSetting'
import ErrorComponent from '@base/Error'
import subscribeFormAction from './subscribeFormAction'
import asyncModuleLoad from './asyncModuleLoad'

const AppWrapper = defineComponent({
  beforeCreate() {
    const {
      appContext: { app },
    } = getCurrentInstance()
    const store = useStore()
    const router = useRouter()

    store.subscribe(({ payload }, state) => {
      subscribeFormAction(payload, router)
    })

    // Handle async module load
    asyncModuleLoad()

    // register global mixin
    app.mixin(globalSetting)
  },
  beforeRouteEnter() {
    console.log('before router enter wrapper')
  },
  errorCaptured(error, component) {},
  setup(props, { slots }) {
    const store = useStore()

    const errorChunkModule = computed(() => store.getters['app/getChunks'])

    return () => {
      return unref(errorChunkModule).onError
        ? h(ErrorComponent, {})
        : slots.default()
    }
  },
})

export default AppWrapper
