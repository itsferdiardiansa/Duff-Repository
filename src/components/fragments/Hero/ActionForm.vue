<template>
  <Form ref="formEl" :model="state.form" @submit="handleSubmit">
    <FormControl
      label="Title"
      :rules="{
        title: [{ required: true, message: 'Title is required' }],
      }"
    >
      <Input
        placeholder="Title"
        name="Title"
        v-model="state.form.title"
        autofocus
      />
    </FormControl>

    <FormControl
      label="Button URL"
      :rules="{
        button_url: [
          { required: true, message: 'Button URL is required' },
          { pattern: urlPattern, message: 'Please check the URL format' },
        ],
      }"
    >
      <Input
        placeholder="Button URL"
        v-model="state.form.button_url"
        autofocus
      />
    </FormControl>

    <FormControl
      label="Description"
      :rules="{
        description: [{ required: true, message: 'Description is required' }],
      }"
    >
      <Textarea name="description" v-model="state.form.description" />
    </FormControl>
    <FormControl
      label="Banner"
      :rules="{
        banner: [{ required: true, message: 'Banner is required' }],
      }"
    >
      <FileUpload name="banner" v-model="state.form.banner" />
    </FormControl>

    <FormControl
      label="Banner Mobile"
      :rules="{
        banner_mobile: [
          { required: true, message: 'Banner Mobile is required' },
        ],
      }"
    >
      <FileUpload name="banner_mobile" v-model="state.form.banner_mobile" />
    </FormControl>

    <FormControl :offset="3">
      <Button
        type="submit"
        variant="orange"
        :label="isCreate ? 'Create' : 'Update'"
        :icon="['fa', 'save']"
        :bold="true"
        :isLoading="isFetching"
      />
    </FormControl>
  </Form>
</template>
<script>
/* eslint-disable */
import { onMounted, reactive, ref, unref } from 'vue';
import Form, { FormControl, FileUpload, Input, Textarea } from '@common/Form';
import Button from '@common/Button';
import { urlPattern } from '@util/pattern';

export default {
  name: 'HeroActionForm',
  components: {
    Button,
    Form,
    FormControl,
    FileUpload,
    Input,
    Textarea,
  },
  emits: ['submit'],
  props: {
    data: {
      type: Object,
      default: () => {},
    },
    isCreate: {
      type: Boolean,
      default: true,
    },
    isFetching: {
      type: Boolean,
      default: false,
    },
    withValidation: {
      type: Boolean,
      default: true,
    },
  },
  setup(props, { emit }) {
    const formEl = ref();
    const state = reactive({
      form: {
        title: '',
        button_url: '',
        banner: '',
        banner_mobile: '',
        position: 0,
        is_active: 0,
        banner_type: 'image',
      },
    });

    const handleSubmit = () => {
      const { withValidation } = props;
      const form = unref(formEl);

      form.validate(valid => {
        if (valid || !withValidation) {
          emit('submit', state.form);
        }
      });
    };

    onMounted(() => {
      const { data, isCreate } = props;

      if (data && !isCreate) state.form = data;
    });

    return {
      formEl,
      state,
      handleSubmit,
      urlPattern,
    };
  },
};
</script>
