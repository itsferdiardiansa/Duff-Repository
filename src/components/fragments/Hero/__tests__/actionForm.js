import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import AppWrapper from '@base/AppWrapper';
import ActionForm from '@fragment/Hero/ActionForm';
import heroModule from '@store/modules/hero';

describe('fragment/Hero/ActionForm', () => {
  let store;

  beforeAll(() => {
    store = createStore({
      modules: {
        hero: heroModule,
      },
    });
  });

  it('will render and match snapshot', () => {
    const Component = mount(<ActionForm />);

    expect(Component.element).toMatchSnapshot();
  });

  it('will render with error validation on all fields and match snapshot', async () => {
    const Component = mount(<ActionForm />);

    await Component.find('form').trigger('submit.prevent');

    const errorFields = Component.findAll(
      `.${Component.rootVM.prefixClass}-control--info`
    );

    expect(errorFields.length).toBe(4);

    await Component.vm.$nextTick(() => {
      expect(Component.element).toMatchSnapshot();
    });
  });

  it('will post data on success and bring up an alert then redirect it to the list', async () => {
    const Component = mount(
      <AppWrapper>
        <ActionForm />
      </AppWrapper>,
      {
        global: {
          plugins: [store],
        },
      }
    );

    const data = { title: 'Hero' };

    await Component.vm.$store.dispatch('hero/postData', {
      data,
      action: 'form.create',
      redirectUrl: '/hero',
    });

    expect(document.body).toMatchSnapshot();
    expect(mockRouterPush).toHaveBeenCalledWith('/hero');
  });

  it('will post data on failed and bring up an alert', async () => {
    const Component = mount(
      <AppWrapper>
        <ActionForm />
      </AppWrapper>,
      {
        global: {
          plugins: [store],
        },
      }
    );

    mockCreateHero.mockReset();
    mockCreateHero.mockRejectedValue({
      message: 'internal.error',
      result: {},
    });

    await Component.vm.$store.dispatch('hero/postData', {
      action: 'form.create',
    });

    expect(document.body).toMatchSnapshot();
  });

  it('will submit by pinning props as a callback', async () => {
    const handleSubmit = jest.fn();
    const formData = {
      title: 'Hero name',
      button_url: 'https://google.com',
      banner: '/9xbg6dsds==', // base64 format
      banner_mobile: '/7343bxx==', // base64 format
    };
    const Component = mount(
      <ActionForm model={formData} onSubmit={handleSubmit} />,
      { props: { withValidation: false } }
    );

    await Component.find('form').trigger('submit.prevent');

    expect(handleSubmit).toHaveBeenCalled();
  });

  it('will render the spinner during the request process', async () => {
    const Component = mount(<ActionForm />, { props: { isFetching: true } });

    expect(
      Component.find(
        `[type="submit"] .${Component.rootVM.prefixClass}-button--spinner`
      ).exists()
    ).toBeTruthy();
    expect(Component.element).toMatchSnapshot();
  });

  it('will render the text and color for the form update', async () => {
    const Component = mount(<ActionForm />, { props: { isCreate: false } });

    expect(Component.find(`[type="submit"]`).text()).toEqual('Update');
    expect(Component.find(`[type="submit"]`).attributes('class')).toContain(
      `${Component.rootVM.prefixClass}-button--warning`
    );
    expect(Component.element).toMatchSnapshot();
  });

  it('will overide local data when in update mode', async () => {
    const formData = {
      title: 'Hero name',
      button_url: 'https://google.com',
      banner: '/9xbg6dsds==', // base64 format
      banner_mobile: '/7343bxx==', // base64 format
    };
    const Component = mount(<ActionForm />, {
      props: { data: formData, isCreate: false },
    });

    expect(Component.vm.state.form).toMatchObject(formData);
    expect(Component.vm.state).toMatchSnapshot();
  });

  it('will execute the form without validation', async () => {
    const handleSubmit = jest.fn();
    const Component = mount(<ActionForm onSubmit={handleSubmit} />, {
      props: { withValidation: false },
    });

    await Component.find('form').trigger('submit.prevent');

    expect(handleSubmit).toHaveBeenCalled();
  });
});
