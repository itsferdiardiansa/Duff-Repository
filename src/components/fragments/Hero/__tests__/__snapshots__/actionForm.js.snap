// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Hero/ActionForm will overide local data when in update mode 1`] = `
Object {
  "form": Object {
    "banner": "/9xbg6dsds==",
    "banner_mobile": "/7343bxx==",
    "button_url": "https://google.com",
    "title": "Hero name",
  },
}
`;

exports[`fragment/Hero/ActionForm will post data on failed and bring up an alert 1`] = `
<body>
  
  
  <div
    id="app"
  >
    

  </div>
  <div
    class="ss-alert-container"
    data-v-app=""
    id="ss-alert__12e5"
    role="alert"
  >
    
    <div
      class="ss-alert ss-alert--success slide-top-enter-from slide-top-enter-active"
    >
      <div
        class="ss-alert--icon"
      >
        <svg
          class=""
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="ss-alert--wrapper"
      >
        <!--v-if-->
        <!--v-if-->
        <div
          class="ss-alert--content"
        >
          <label>
            cms.hero.create.success
          </label>
        </div>
      </div>
    </div>
    
  </div>
</body>
`;

exports[`fragment/Hero/ActionForm will post data on success and bring up an alert then redirect it to the list 1`] = `
<body>
  
  
  <div
    id="app"
  >
    

  </div>
  <div
    class="ss-alert-container"
    data-v-app=""
    id="ss-alert__12e5"
    role="alert"
  >
    
    <div
      class="ss-alert ss-alert--success slide-top-enter-from slide-top-enter-active"
    >
      <div
        class="ss-alert--icon"
      >
        <svg
          class=""
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="ss-alert--wrapper"
      >
        <!--v-if-->
        <!--v-if-->
        <div
          class="ss-alert--content"
        >
          <label>
            cms.hero.create.success
          </label>
        </div>
      </div>
    </div>
    
  </div>
</body>
`;

exports[`fragment/Hero/ActionForm will render and match snapshot 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Title
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="Title"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="Title"
          placeholder="Title"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Button URL
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="ldp_id"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="ldp_id"
          placeholder="Button URL"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner Mobile
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner_mobile"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner_mobile"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--dark ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Create
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Hero/ActionForm will render the spinner during the request process 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Title
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="Title"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="Title"
          placeholder="Title"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Button URL
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="ldp_id"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="ldp_id"
          placeholder="Button URL"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner Mobile
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner_mobile"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner_mobile"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--dark ss-button--base text-bold"
        disabled=""
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <!--v-if-->
          
          
          <div
            class="ss-button--spinner"
          >
            <div
              class="ss-spinner"
              style="width: 20px; height: 20px;"
            >
              <div />
              <div />
              <div />
            </div>
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Hero/ActionForm will render the text and color for the form update 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Title
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="Title"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="Title"
          placeholder="Title"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Button URL
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="ldp_id"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="ldp_id"
          placeholder="Button URL"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <div
      class="ss-form-control--label"
    >
      Banner Mobile
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner_mobile"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner_mobile"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--warning ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Update
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;

exports[`fragment/Hero/ActionForm will render with error validation on all fields and match snapshot 1`] = `
<form
  class="ss-form"
>
  
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Title
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="Title"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="Title"
          placeholder="Title"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Title is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Button URL
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        autofocus=""
        class="ss-control--input"
        name="ldp_id"
      >
        <!--v-if-->
        <input
          autocomplete="off"
          autofocus=""
          class="ss-input"
          name="ldp_id"
          placeholder="Button URL"
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Button URL is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Banner
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Banner is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control has-feedback"
  >
    <div
      class="ss-form-control--label"
    >
      Banner Mobile
    </div>
    <div
      class="ss-form-control--wrapper"
    >
      <div
        class="ss-control--file"
        name="banner_mobile"
      >
        <input
          accept="image/png, image/jpeg"
          name="banner_mobile"
          type="file"
        />
        <!--v-if-->
        
        <div
          class="file-hero"
        >
          <svg
            class=""
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M296 384h-80c-13.3 0-24-10.7-24-24V192h-87.7c-17.8 0-26.7-21.5-14.1-34.1L242.3 5.7c7.5-7.5 19.8-7.5 27.3 0l152.2 152.2c12.6 12.6 3.7 34.1-14.1 34.1H320v168c0 13.3-10.7 24-24 24zm216-8v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h136v8c0 30.9 25.1 56 56 56h80c30.9 0 56-25.1 56-56v-8h136c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <label>
            Click or drag to upload
          </label>
        </div>
        
      </div>
      <div
        class="ss-control--info slide-top-enter-from slide-top-enter-active"
      >
        Banner Mobile is required
      </div>
    </div>
  </div>
  <div
    class="ss-form-control"
  >
    <!---->
    <div
      class="ss-form-control--wrapper col-offset-3"
    >
      <button
        class="ss-button ss-button--dark ss-button--base text-bold"
        type="submit"
      >
        <div
          class="ss-button--wrapper"
        >
          
          <div
            class="ss-button--icon"
          >
            <svg
              class=""
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
                fill="currentColor"
              />
            </svg>
          </div>
          
          
          <div
            class="ss-button--label has-icon"
          >
            
            Create
            
          </div>
          
        </div>
      </button>
      <!---->
    </div>
  </div>
  
</form>
`;
